component/main.jsonnet

```
@@ -28,19 +30,32 @@ local secrets = {
     DB_VENDOR: params.database.external.vendor,
     DB_ADDR: params.database.external.host,
     DB_PORT: std.toString(params.database.external.port),
+    [if params.database.tls.enabled then 'JDBC_PARAMS']: params.database.tls.jdbcParams,
   },
 };
 
 local db_secret = kube.Secret(params.database.secretname) {
   metadata+: {
     labels+: params.labels,
   },
-  stringData: secrets[params.database.provider],
+  stringData: connection_secrets[params.database.provider],
+};
+
+local db_cert_secret = kube.Secret(params.database.tls.certSecretName) {
+  metadata+: {
+    labels+: params.labels,
+  },
+  type: 'kubernetes.io/tls',
+  stringData: {
+    'tls.key': params.database.tls.serverCertKey,
+    'tls.crt': params.database.tls.serverCert,
+  },
 };
 
 // Define outputs below
 {
   '00_namespace': namespace,
   '10_admin_secret': admin_secret,
   '11_db_secret': db_secret,
+  [if params.database.tls.enabled && params.database.tls.verification != 'verify' then '13_db_certs']: db_cert_secret,
```

If you want to always mount the db certs secret as a volume:

```suggestion
 '13_db_certs': db_cert_secret,
```